src/nodes/NodeConnectionManager.ts

```diff
@@ -651,21 +692,21 @@ class NodeConnectionManager {
   )
   protected async getConnection(
     targetNodeId: NodeId,
-    address: NodeAddress | undefined,
+    addresses: Array<NodeAddress> | undefined,
     @context ctx: ContextTimed,
   ): Promise<ConnectionAndTimer> {
     // If the connection already exists then we need to return it.
     const existingConnection = await this.getExistingConnection(targetNodeId);
     if (existingConnection != null) return existingConnection;
 
-    // If there was no address provided then we need to find it.
-    if (address == null) {
+    // If there were no addresses provided then we need to find them.
+    if (addresses == null) {
```

Comment:
Since `addresses` is an array now, you need to check if it's an empty array and handle that. It should throw an error as invalid usage. 

It's probably checked somewhere else so likely won't cause a problem. But as the saying goes, fail early and fail often.

Reply:
```ts
// If there were no addresses provided then we need to find them.
    if (addresses == null || addresses.length === 0) {
      // Find the node
      addresses = await this.findNodeAll(targetNodeId, undefined, ctx);
      if (addresses.length === 0) throw new nodesErrors.ErrorNodeGraphNodeIdNotFound();
    }
```